src/storage/stat.cc

```diff
@@ -0,0 +1,106 @@
+#include "stat.h"
+
+#include <fcntl.h>
+#include <sys/statvfs.h>
+#include <unistd.h>
+#include <cmath>
+#include <cstring>
+#include <sstream>
+#include <stdexcept>
+#include <tuple>
+
+namespace storage {
+
+struct Stat {
+  uint64_t blockSize;
+  uint64_t freeBlockNumber;
+  uint64_t blockNumb;
+};
+
+static std::string getStat(const std::string& path, Stat& stat) {
+  int fd{-1};
+  if (-1 == (fd = open(path.c_str(), O_DIRECTORY | O_RDONLY))) {
+    return std::string("Failed to open a file/directory; path: ") + path + ", err: " + std::strerror(errno);
+  }
+  struct statvfs fs_stat {};
+  const auto stat_res = fstatvfs(fd, &fs_stat);
+  close(fd);
+  if (-1 == stat_res) {
+    return std::string("Failed to obtain statistic about the path volume; path: ") + path +
+           ", err: " + std::strerror(errno);
+  }
+  stat.blockNumb = fs_stat.f_blocks;
+  stat.freeBlockNumber = (getuid() != 0 ? fs_stat.f_bavail : fs_stat.f_bfree);
```

Comment:
is this for unit tests? Don't think we can run as non-root? maybe this could use a comment.

Reply:
Yes. This makes code user permission agnostic, actually I copy-pasted it from the libostree :).